### PR TITLE
Harris week 5

### PR DESCRIPTION
**Pull request by HarrisSheraz20**
# Your Details

Your Name:
Your City:
Your Slack Name:

# Homework Details

Module:
Week:

### CODE REVIEW

**Comment by ahmad-ali14** [comment]
@HarrisSheraz20  could you please specify  which week and module this PR for

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
 }
 
+
 function greaterThan10() {
   let num = 10;
-  let isBigEnough;
-
-  if (isBigEnough) {
+  let isBigEnough = num;
+  if (isBigEnough >= num) {
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/1-fix-functions.js:18]: Again `semantics`.
- you assign `isBigEnough` to `num`
- then you compare them together, this comparison will always return true. ( think of the reason )
- I would rather solve it as follows: 
```
 let num = 10;
 let isBigEnough;

  if (isBigEnough >= 10 {
          ...
```

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
     return "num is greater than or equal to 10";
   } else {
     return "num is not big enough";
   }
 }
 
+
 function sortArray() {
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/1-fix-functions.js:26]: good

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
+  sortedLetters = letters.sort();
   return sortedLetters;
 }
 
+
 function first5() {
   let numbers = [1, 2, 3, 4, 5, 6, 7, 8];
   let sliced;
-
+  sliced = numbers.slice(0, 5);
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/1-fix-functions.js:37]: we need the first 5, we started slicing from 0.
`0 , 1 , 2 , 3, 4 `
how many numbers are there ??

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 4]
```diff
@@ -1,45 +1,47 @@
...
   return sliced;
 }
 
+
 function get3rdIndex(arr) {
   let index = 3;
-  let element;
-
+  let element = arr[index];
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/1-fix-functions.js:44]: we started from 0, not 1.

File: week-2/Homework/mandatory/2-function-creation.js [hunk 5]
```diff
@@ -5,7 +5,15 @@ Write a function that:
 - removes any forward slashes (/) in the strings
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+
+function tidyUpString(strArr) {
+  for (let i = 0; i < strArr.length; i++) {
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/2-function-creation.js:10]: - you are returning strArr1, which is undefined. I think it's a typo.

Comment (ahmad-ali14) [reply week-2/Homework/mandatory/2-function-creation.js:10]: Good job

File: week-2/Homework/mandatory/2-function-creation.js [hunk 6]
```diff
@@ -15,8 +23,13 @@ Complete the function to check if the variable `num` satisfies the following req
 Tip: use logical operators
 */
 
-function validate(num) {}
-
+function validate(num) {
+  if (typeof num === 'number' && num % 2 === 0 && num <= 100){
+    return true
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/2-function-creation.js:28]: perfect !!

File: week-2/Homework/mandatory/2-function-creation.js [hunk 7]
```diff
@@ -26,21 +39,36 @@ The function must:
 */
 
 function remove(arr, index) {
-  return; // complete this statement
+  let newArr = arr.slice()
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/2-function-creation.js:42]: - clever idea to use `slice()` to copy the array.
- there is easier way: ` let newArr = [ ...arr ] `

File: week-2/Homework/mandatory/2-function-creation.js [hunk 8]
```diff
@@ -26,21 +39,36 @@ The function must:
...
+
+  for (let i = 0; i < arr.length; i++) {
+
+    if (arr[i]> 100) {arr[i] = 100}
+    arr[i] = Math.round(arr[i]*100)/100;
+    arr[i] = arr[i].toString();
+    arr[i] = arr[i].concat("%");
+
+  } 
+ return arr
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/2-function-creation.js:67]: perfect

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 9]
```diff
@@ -5,38 +5,50 @@
...
+task 2
+  3. What is printed to the console?  // 2, 6, 4, 9, 6, 13, 8   
+  console.log(x);   2
+  console.log(a);   6
+  console.log(d);   4, 6, 8 
+  console.log(e);   9, 13
+
+  4. How many times is "f1" called?   // 1st call of f1 is when consol.log(e) = 9,  //  last time f1 is called and resulting in console.log(e) = 13  
+  5. How many times is "f2" called?  // 1st call of f2 is when console.log(d) = 4, second time f2 is called and resulting in console.log(d) = 6   // last time f2 is called and resulting in console.log(d) = 8
+  6. What value does the "a" parameter take in the first "f1" call?        // value 8
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/3-playing-computer.js:22]: - the first call for `f1` when `i =1`, 
- we are calling the function by passing `f1( i , a)`
- then parameter `a` takes `i = 1`.
- parameter `b` takes `a=6`.

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 10]
```diff
@@ -5,38 +5,50 @@
...
+  console.log(x);   2
+  console.log(a);   6
+  console.log(d);   4, 6, 8 
+  console.log(e);   9, 13
+
+  4. How many times is "f1" called?   // 1st call of f1 is when consol.log(e) = 9,  //  last time f1 is called and resulting in console.log(e) = 13  
+  5. How many times is "f2" called?  // 1st call of f2 is when console.log(d) = 4, second time f2 is called and resulting in console.log(d) = 6   // last time f2 is called and resulting in console.log(d) = 8
+  6. What value does the "a" parameter take in the first "f1" call?        // value 8 
   7. What is the value of the "a" outer variable when "f1" is called for the first time?
 */
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/3-playing-computer.js:24]: here a is outside the function so it takes its normal value `a = 6`

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 11]
```diff
@@ -5,38 +5,50 @@
...
+task 1 
+  1. This program throws an error. Why? (If you can't find it, try executing it).  //because "b" is not defined because it had no varruble with the value./
   2. Remove the line that throws the error.
-  3. What is printed to the console?
-  4. How many times is "f1" called?
-  5. How many times is "f2" called?
-  6. What value does the "a" parameter take in the first "f1" call?
+
+task 2
+  3. What is printed to the console?  // 2, 6, 4, 9, 6, 13, 8
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/3-playing-computer.js:14]: execute the program again.

File: week-2/Homework/mandatory/4-sorting-algorithm.js [hunk 12]
```diff
@@ -14,9 +14,31 @@ You don't have to worry about making this algorithm work fast! The idea is to ge
 "think" like a computer and practice your knowledge of basic JavaScript.
 */
 
+
 function sortAges(arr) {
```
Comment (ahmad-ali14) [inline week-2/Homework/mandatory/4-sorting-algorithm.js:18]: perefect

**Comment by ahmad-ali14** [comment]
Java script week - 2 reviewed.

File: week-2/Homework/mandatory/3-playing-computer.js (hunk 9 above)
Comment (ahmad-ali14) [reply week-2/Homework/mandatory/3-playing-computer.js:22]: execute this program here http://pythontutor.com/visualize.html#mode=edit
and see how things are going step by step

**Comment by ahmad-ali14** [comment]
@HarrisSheraz20 if you want we can have like 1-hour session on zoom to discuss this review

**Comment by HarrisSheraz20** [comment]
hello Ahmad,

yes sorry that it took me this long to respond to your emails. because I
was a bit busy, so yes I am in the need of help with some work and I would
like to work on the feedback you asked me, so what time today are you
available for a call on slack

**Pull request closed without merging**